platform/macos/display_server_macos.mm

```diff
@@ -2573,7 +2573,13 @@
 				[wd.window_object setContentMaxSize:NSMakeSize(size.x, size.y)];
 			}
 			[wd.window_object toggleFullScreen:nil];
+
+			if (p_mode == WINDOW_MODE_EXCLUSIVE_FULLSCREEN) {
```

Comment:
Isn't this later override in the `switch (p_mode)`? Or it's needed to force going back to default presentation for some updates before going exclude fullscreen?

Reply:
Should be `old_mode`.

Reply:
Right, I thought about that too but then wondered why it would only be handling this for `WINDOW_MODE_FULLSCREEN` and not for the other non-fullscreen modes. But those are handled by `windowDidExitFullScreen` I suppose.